### PR TITLE
Side menu

### PR DESCRIPTION
**Pull request by iamtarun1993**

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,22 +1,49 @@
...
+    <ion-header class="sidemenu--header">
+      <ion-toolbar class="sidemenu--header-info-wrapper" translucent>
+        <ion-grid>
+          <ion-row class="sidemenu--header-wrapper">
+          <ion-col size="2">
+            <img class="sidemenu--header-logo" src="assets/svg/logo-icon-white.svg">
+          </ion-col>
+          <ion-col size="10">
+            <div class="sidemenu--info-wrapper">
+              <p class="sidemenu--user-name">{{ this.eou && this.eou.us.full_name }}</p> <!-- Need to add characters pipe later -->
```
Comment (Julias0) [inline src/app/app.component.html:12]: this not required here
eou?.us?. full_name <----- this should work

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,22 +1,49 @@
...
+              <p class="sidemenu--org-name">{{ this.activeOrg && this.activeOrg.name }}</p>
```
Comment (Julias0) [inline src/app/app.component.html:13]: same format as above

File: src/app/app.component.ts [hunk 3]
```diff
@@ -60,10 +51,183 @@ export class AppComponent implements OnInit {
...
+    );
+    const orgUserSettings$ = this.orgUserSettingsService.get();
+    const delegatedAccounts$ = this.orgUserService.findDelegatedAccounts();
+    const deviceInfo$ = this.deviceService.getDeviceInfo();
+
+    let allowedReportsActions: any;
+    let allowedAdvancesActions: any;
+    let allowedTripsActions: any;
+
+    delegatedAccounts$.pipe(
```
Comment (Julias0) [inline src/app/app.component.ts:78]: this pipe is not connected to anything - why is this here?

File: src/app/app.component.ts [hunk 4]
```diff
@@ -60,10 +51,183 @@ export class AppComponent implements OnInit {
...
+      orgs: orgs$,
+      currentOrg: currentOrg$,
+      orgSettings: orgSettings$,
+      orgUserSettings: orgUserSettings$,
+      delegatedAccounts: delegatedAccounts$,
+      allowedActions: allowedActions$,
+      deviceInfo: deviceInfo$
+    });
+
+    primaryData$.subscribe((res) => {
```
Comment (Julias0) [inline src/app/app.component.ts:113]: directly subscribe to above

File: src/app/app.component.ts [hunk 5]
```diff
@@ -60,10 +51,183 @@ export class AppComponent implements OnInit {
...
+    const orgs$ = this.offlineService.getOrgs();
+    const currentOrg$ = this.offlineService.getCurrentOrg();
+    const orgSettings$ = this.offlineService.getOrgSettings().pipe(
+      shareReplay()
+    );
+    const orgUserSettings$ = this.orgUserSettingsService.get();
+    const delegatedAccounts$ = this.orgUserService.findDelegatedAccounts();
+    const deviceInfo$ = this.deviceService.getDeviceInfo();
+
+    let allowedReportsActions: any;
```
Comment (Julias0) [inline src/app/app.component.ts:74]: are these being used?

File: src/app/app.component.ts [hunk 6]
```diff
@@ -60,10 +51,183 @@ export class AppComponent implements OnInit {
...
+      const orgSettings = res.orgSettings;
+      const orgUserSettings = res.orgUserSettings;
+      const isDelegatee = res.delegatedAccounts.length > 0;
+      this.appVersion = (res.deviceInfo && res.deviceInfo.appVersion) || '1.2.3';
+      allowedReportsActions = res.allowedActions && res.allowedActions.allowedReportsActions;
+      allowedAdvancesActions = res.allowedActions && res.allowedActions.allowedAdvancesActions;
+      allowedTripsActions = res.allowedActions && res.allowedActions.allowedTripsActions;
+      let isSwitchedToDelegator = false;
+
+      this.orgUserService.isSwitchedToDelegator().then((res) => {
```
Comment (Julias0) [inline src/app/app.component.ts:126]: add this to the above forkJoin
convert this to observable - isSwitchedToDelegator

Comment (Julias0) [reply src/app/app.component.ts:126]: If not possible you can do it by - 

isSwitchedToDelegator$ = from(this.orgUserService.isSwitchedToDelegator())

File: src/app/app.component.ts (hunk 3 above)
Comment (iamtarun1993) [reply src/app/app.component.ts:78]: this pipe is connected to 
`delegatedAccounts: delegatedAccounts$,`

File: src/app/app.component.html [hunk 7]
```diff
@@ -1,22 +1,49 @@
...
+    <ion-header class="sidemenu--header">
+      <ion-toolbar class="sidemenu--header-info-wrapper" translucent>
+        <ion-grid>
+          <ion-row class="sidemenu--header-wrapper">
+          <ion-col size="2">
+            <img class="sidemenu--header-logo" src="assets/svg/logo-icon-white.svg">
+          </ion-col>
+          <ion-col size="10">
+            <div class="sidemenu--info-wrapper">
+              <p class="sidemenu--user-name">{{ eou?.us?. full_name }}</p> <!-- Need to add characters pipe later -->
```
Comment (Julias0) [inline src/app/app.component.html:12]: no space between . and variable name

File: src/app/core/services/permissions.service.ts [hunk 8]
```diff
@@ -2208,7 +2209,8 @@ export class PermissionsService {
     return filteredRoles$.pipe(
       map(
         filteredRoles => {
-          if (this.allowedAccess(resource, orgSettings)) {
+          try{
```
Comment (Julias0) [inline src/app/core/services/permissions.service.ts:2212]: remove try catch

**Review by Julias0 — APPROVED** [review]

File: src/app/app.component.html [hunk 9]
```diff
@@ -1,22 +1,49 @@
...
+      <ion-toolbar class="sidemenu--header-info-wrapper" translucent>
+        <ion-grid>
+          <ion-row class="sidemenu--header-wrapper">
+          <ion-col size="2">
+            <img class="sidemenu--header-logo" src="assets/svg/logo-icon-white.svg">
+          </ion-col>
+          <ion-col size="10">
+            <div class="sidemenu--info-wrapper">
+              <p class="sidemenu--user-name">{{ eou?.us?.full_name }}</p> <!-- Need to add characters pipe later -->
+              <p class="sidemenu--org-name">{{ activeOrg?.name }}</p>
```
Comment (madhavmansuriya40) [inline src/app/app.component.html:13]: here as well (some orgs have long names)
